### ISSUE TITLE
Callback url causes 404 page to be displayed

### ISSUE THREAD

**Issue by justin-barca-at-camis** [authts/react-oidc-context#310]
In the documentation, it is stated 

```
You still need to setup a redirect uri, which must point to your application, but you do not need to create that route.
```

I found this misleading because in the common case of using `react-router-dom@6` with a 404 catch-all route, the 404 page is displayed when the code exchange callback route is navigated to.

Should we not encourage the user to set up a route like so?
```
 <Route path={ROUTES.CALLBACK} element={<LoadingIndicator />} />
```

**Comment by pamapa** [comment]
Good point, was not aware of that `react-router-dom@6` feature. The route you show above is probably too limiting for everybody. Maybe it enough to extend the sentence:

> You still need to setup a redirect uri, which must point to your application, but you do not need to create that route.

You still need to setup a redirect uri, which must point to your application. Unless you have not setup a `404 catch-all route`
behavior,  you do not need to create that route.

**Comment by justin-barca-at-camis** [comment]
To anyone coming up on this issue, a quick tip: make sure to do a browser history replace and not push when you advance from the callback URL, otherwise your history will move back to it unexpectedly.

**Comment by kodeschooler** [comment]
Can someone please point me to some documentation for this? For example, do I not need protected routes with this? Is the suggestion to use useAuth in each component that requires you to be logged in and it'll just work? I'm currently using the react-oidc library, but it's not being maintained and the documentation is poor so I am thinking of switching to this.  

If I have a component that makes api calls, and if the user has had the page open all night and comes back to click on a button to fetch some data, will I get a 401 due to inactivity? Or is it supposed to continuously get new access tokens? I'm just wondering what options I have and if I can get some visibility into what's happening even on a high level. 


Why am I getting "Oops... no matching state found in storage" after clicking login (using the example in the readme)? I can't find that text anywhere in the source. 

I'm trying to contrast this with oidc-react. That library has you create the UserManager class and pass it to the context. What does this one do that oidc-react doesn't? Or vice versa. 

Thanks

**Comment by pamapa** [comment]
This library creates the `UserManager` inside. At the end there is not much code in this library you, best is you read it directly. A keep part of this library can be seen [here](https://github.com/authts/react-oidc-context/blob/main/src/AuthProvider.tsx#L154-L164): If auth parameters are detected it acts accordingly.

- In order to check if the user has logged in or not you need to use `auth.isAuthenticated`.
- When calling an api which needs the access token, you need to inject that access token in your fetch. As documented [here](https://github.com/authts/react-oidc-context#call-a-protected-api)

**Comment by kodeschooler** [comment]
This is the lib the auth context is coming from: https://github.com/bjerkio/oidc-react
It returns this: 

```
export interface AuthContextProps {
    signIn: (args?: unknown) => Promise<void>;
    signInPopup: () => Promise<void>;
    signOut: () => Promise<void>;
    signOutRedirect: (args?: unknown) => Promise<void>;
    userManager: UserManager;
    userData?: User | null;
    isLoading: boolean;
}
```

It's based on the oidc-client.js (the predecessor to oidc-client-ts). I think what might be happening in my scenario is auth.userData is gone (but why I don't know) so it never adds the token to the request, which would cause a 401. Then the error is caught in the interceptor and it does the auth.signIn(). 

I could decouple the useAuth from the hook, but then every component would need to know about access tokens. 

Maybe I should just switch to this one as it seems to have more support and its api exposes more. I just need to sell it to my boss.  Where is an example of how to enable logging? I went here and I don't see it: https://www.npmjs.com/package/react-oidc-context#documentation

Can you show an example, because you're right- I really do need logging to know what's really going on.

**Comment by pamapa** [comment]
You can enable the logging of `oidc-client-ts`. See here https://authts.github.io/oidc-client-ts/#logging

**Comment by bwash70** [comment]
I'm getting this 404 issue as well. It works fine on localhost but once I publish the app and the user authenticates, it redirects me to a 404 page.

I'm relatively inexperienced so I'm having a bit of trouble understanding how you fixed it?

I have updated and triple checked the authorised redirect URLs, I don't have a catch all for 404 errors, I'm using react-dom-router v6, and I have the route: <Route path="/linkedin" element={<LinkedInCallback />} />.

Any help you could give me would be great - I've been at this for a bunch of hours and I'm running out of ideas.
